apple/internal/macos_rules.bzl

```diff
@@ -1253,24 +1274,17 @@ def _macos_dylib_impl(ctx):
         bundle_post_process_and_sign = False,
         entitlements = entitlements,
         executable_name = executable_name,
-        partials = [debug_outputs_partial],
+        partials = [debug_outputs_partial, partial.make(_dylib_partial, binary_artifact)],
         platform_prerequisites = platform_prerequisites,
         predeclared_outputs = predeclared_outputs,
         provisioning_profile = getattr(ctx.file, "provisioning_profile", None),
         rule_descriptor = rule_descriptor,
         rule_executables = rule_executables,
         rule_label = label,
     )
-    output_file = actions.declare_file(label.name + ".dylib")
-    codesigning_support.sign_binary_action(ctx, binary_artifact, output_file)
 
     return [
-        AppleBinaryInfo(
-            binary = output_file,
-            product_type = rule_descriptor.product_type,
-        ),
         DefaultInfo(files = depset(transitive = [
-            depset([output_file]),
             processor_result.output_files,
         ])),
         OutputGroupInfo(
```

Comment:
I think you could inline the dylib into this, if you had to you could do the merging with a dictionary of `{"dylib": output_file}` or something here? If this worked this would be a less invasive change